documentation/book/con-upgrading-kafka-versions.adoc

```diff
@@ -0,0 +1,62 @@
+* The required version of Zookeeper.
+
+The differences in these attributes between the Kafka version being upgraded from and the Kafka version being upgraded to will determine exactly how the upgrade process works.
+
+== When all the attributes are the same
+
+When all these properties are the same for the Kafka version being upgraded from and to then when you change the  `Kafka.spec.kafka.config` a single rolling update of the Kafka brokers can be used by the Cluster Operator.
+This will typically be the case when the Kafka version numbers differ only in the patch level, for example 2.0.0 to 2.0.1.
+
+== When the interbroker protocol versions differ
+
+When the Kafka versions use different versions of the Kafka protocol the Cluster Operator will automatically handle the setting and unsetting of the `interbroker.protocol.version` together with the change in the `Pod.spec.image`.
+Doing this requires two rolling updates of the broker cluster.
+
+== When the message format versions differ
+
+When the Kafka versions use different versions for the message format in the Kafka logs you will need to configure the `log.message.format.version` in the `Kafka.spec.kafka.config` explicitly:
+
+. Initially it will need to be set to the message format version being used in the Kafka version being upgraded from. 
+If this wasn't explicitly configured already the Cluster Operator will perform a rolling update when you change this, as if would for any other `Kafka.spec.kafka.config` change.
+
+. You can then set the `Kafka.spec.kafka.version` to the version of Kafka being upgraded to, keeping the `log.message.format.version` set to the old version.
+The Cluster Operator will perform the rolling upgrades necessary to start using the new Kafka version.
+
+. Once the Cluster Operator has completed its rolling updates you will need for upgrade client applications to use the new version of the Kafka client libraries. 
+
+. When you've upgraded all the client applications and you are sure you want to proceed with the new version, you can change the `log.message.format.version` in the `Kafka.spec.kafka.config` to the new version.
++
+NOTE: Equivalently, you can remove `log.message.format.version` from the `Kafka.spec.kafka.config`, however that will mean the next Kafka upgrade will require an initial rolling update in order to set `log.message.format.version` explicitly.
++
+WARNING: Once you have changed `log.message.format.version` from the old version it will no longer be possible to revert to the old version, so it is worth ensuring that you are ready to proceed
++
+The Cluster Operator will perform a rolling upgrade of the broker cluster so that all brokers are using the new format for messages.
+
+== When the versions of Zookeeper differ
+
+When the Kafka versions need a different version of Zookeeper then an upgrade of Zookeeper is also be necessary.
+Currently all Kafka versions support the same version of Zookeeper. 
```

Comment:
Do we need to make it clear that all Kafka versions supported by {ProductName} use the same Zookeeper version?